doc/mgmt/SONiC_PIM_SM_HLD.md

```diff
@@ -0,0 +1,1036 @@
+| (*,G) or (S,G) OIF as Vlan10 | (*,G) or (S,G) OIF as Vlan10,pimreg | SET | SAI_PACKET_ACTION_COPY |
+| (*,G) or (S,G) OIF as Vlan10,pimreg | (*,G) or (S,G) OIF as Vlan10 | SET | SAI_PACKET_ACTION_COPY_CANCEL |
+| (*,G) or (S,G) OIF as Vlan10,pimreg | (*,G) or (S,G) OIF as Vlan10,pimreg | DEL | IPMC entry removed |
+| (*,G) or (S,G) OIF as pimreg | (*,G) or (S,G) OIF as pimreg | DEL | IPMC entry removed |
+
+#### 3.1.2.2 IIF expansion in OIF List
+
+IIF member ports to be included in IPMC group member on following cases,
+1. L3 multicast case, local receiver on the IIF vlan then add IIF vlan in OIF list
+2. Shared case (L3 and Snooping enabled), local receiver on the IIF vlan then add IIF vlan member ports into OIF list
+
+FRR will remove the IIF in OIF list for SG entries. So APPL_DB will have empty OIF list.
+
+Configuration sequence,
+1. FRR sends update to ipmcfpmsyncd with IIF as Vlan and OIF as empty
+2. ipmcfpmsyncd populates S,G with IIF as VLAN and OIF empty in APPL_DB
+3. ipmcorchagent, upon receiving the update adds IIF VLAN to OIF list(class MRouteEntry and member list m_oifs updated with IIF vlan)<br>
+    a) calls IPMC group creation incase of L3 VLAN <br>
+    b) call L2MC group creation incase of snooping enabled on the VLAN <br>
+4. SAI expands the L2/L3 portbitmap for IIF/OIF Vlans
+
+| Source IP       | Group IP        | Incoming Interface   |Outgoing Interface(s)
+|----------------|----------------|--------------------| ---------------------
+|30.0.0.3 |        225.1.1.1|        Vlan300|              Vlan100
+| | | |    Vlan300
+|30.0.0.4 |        225.1.1.1|        Vlan300|              Vlan300
+
+#### 3.1.2.3 IPMC Aging time
+
+IPMC aging timeout is switch attribute for SAI. This is global configuration for a switch and applicable to all VRF, all AFI's.
+
+Following the sequence of configuration,
+1. FRR sends update to ipmcfpmsyncd with aging value using struct mfpm_kat_info_t.
+2. ipmcfpmsyncd populate "ipmc_aging_time" entry in  SWITCH_TABLE (APPL_DB)
+3. ipmcorchagent, receives SWITCH_TABLE (APPL_DB) update and calls set_switch_attribute with attr as SAI_SWITCH_ATTR_IPMC_AGING_TIME and value "ipmc_aging_time" <br>
+
+### 3.1.3 Syncd changes
+#### 3.1.3.1 Changes required to process IPMC entries AGE and DONTAGE data
+Following will be added in the syncd inorder to process the IPMC entry or entries AGE and DONTAGE data,
+
+1. Establish a connection with STATE_DB inorder to add or remove IPMC entries AGE and DONTAGE data.
+2. New notification handler (Notificationhandler::onIpmcEvent) API will be defined and registered with SAI to receive IPMC specific AGE and
+   DONTAGE notifications.
+   IPMC specific notification will be identified with the enum "SAI_SWITCH_NOTIFICATION_NAME_IPMC_EVENT" which will be added 
+   newly in SAI layer. 
+3. On receiving the IPMC entry/entries AGE or DONTAGE data, required validations will be done after 
```

Comment:
Hope both SAI and syncd handled bulk update handling for KAT update

Reply:
My understand is that whether SAI and SYNCD are designed to handle multiple IPMC entries aliveness data notifications? 

Yes, both SAI and SYNCD are designed to handle IPMC entries bulk data similar to FDB entries handling.  As part of callback, SAI provides the notification count and data array.